kafka-server/src/test/java/com/ozangunalp/kafka/server/BrokerConfigTest.java

```diff
@@ -61,6 +61,26 @@ void testOverrideListeners() {
         assertThat(properties).containsEntry(KafkaConfig.ListenerSecurityProtocolMapProp(), "SSL:SSL,CONTROLLER:PLAINTEXT");
     }
 
+    @Test
+    void testKraftBrokerRoleOnly() {
+        Properties props = new Properties();
+        props.put(KafkaConfig.ProcessRolesProp(), "broker");
+        props.put(KafkaConfig.BrokerIdProp(), "2");
+        props.put(KafkaConfig.QuorumVotersProp(), "1@:9094");
+
+        Properties properties = BrokerConfig.defaultCoreConfig(props, "", 9092, 9093, 9094, PLAINTEXT);
+
+        assertThat(properties).containsEntry(KafkaConfig.BrokerIdProp(), "2");
+        assertThat(properties).containsEntry(KafkaConfig.QuorumVotersProp(), "1@:9094");
+        assertThat(properties).containsEntry(KafkaConfig.ListenersProp(), "BROKER://:9093,PLAINTEXT://:9092");
+        assertThat(properties).containsEntry(KafkaConfig.ProcessRolesProp(), "broker");
+        assertThat(properties).containsEntry(KafkaConfig.ControllerListenerNamesProp(), "CONTROLLER");
+        assertThat(properties).containsEntry(KafkaConfig.InterBrokerListenerNameProp(), "BROKER");
+        assertThat(properties).containsEntry(KafkaConfig.AdvertisedListenersProp(), "PLAINTEXT://:9092,BROKER://:9093");
+        assertThat(properties).containsEntry(KafkaConfig.EarlyStartListenersProp(), "BROKER");
+        assertThat(properties).containsEntry(KafkaConfig.ListenerSecurityProtocolMapProp(), "BROKER:PLAINTEXT,CONTROLLER:PLAINTEXT,PLAINTEXT:PLAINTEXT");
```

Comment:
The broker in broker role needs to be able to connect to the controller(s).  It needs the listener security property map and the listener names prop to allow it to do that.